independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanGenerator.java

```diff
@@ -379,6 +379,7 @@ public BeanGenerator(AnnotationLiteralProcessor annotationLiterals, Predicate<Do
         }
         implementGetBeanClass(bean, beanCreator);
         implementGetName(bean, beanCreator);
+        implementIsDefaultBean(bean, beanCreator);
```

Comment:
We should only implement the method for default beans - it's a default method for others.

Reply:
Makes sense, updated